app/src/main/java/de/tum/in/tumcampusapp/services/SilenceService.java

```diff
@@ -96,47 +98,44 @@ protected void onHandleWork(@NonNull Intent intent) {
+        List<CalendarItem> currentLectures = calendarManager.getCurrentFromDb();
+        Utils.log("Current lectures: " + currentLectures.size());
+
+        if (currentLectures.size() == 0 || isDoNotDisturbMode()) {
+            if (Utils.getInternalSettingBool(this, Const.SILENCE_ON, false) && !isDoNotDisturbMode()) {
+                // default: old state
+                Utils.log("set ringer mode to old state");
+                am.setRingerMode(Integer.parseInt(
+                        Utils.getSetting(this, Const.SILENCE_OLD_STATE,
+                                         Integer.toString(AudioManager.RINGER_MODE_NORMAL))));
+                Utils.setInternalSetting(this, Const.SILENCE_ON, false);
+
+                List<CalendarItem> nextCalendarItem = calendarManager.getNextCalendarItem();
+                if (nextCalendarItem.size() != 0) { //Check if we have a "next" item in the database and update the refresh interval until then. Otherwise use default interval.
+                    // refresh when next event has started
+                    waitDuration = getWaitDuration(nextCalendarItem.get(0).getDtstart());
                 }
+            }
+        } else {
+            // remember old state if just activated ; in doubt dont change
+            if (!Utils.getInternalSettingBool(this, Const.SILENCE_ON, true)) {
+                Utils.setSetting(this, Const.SILENCE_OLD_STATE, am.getRingerMode());
+            }
 
-                // if current lecture(s) found, silence the mobile
-                Utils.setInternalSetting(this, Const.SILENCE_ON, true);
-
-                // Set into silent mode
-                String mode = Utils.getSetting(this, "silent_mode_set_to", "0");
-                if ("0".equals(mode)) {
-                    Utils.log("set ringer mode: vibration");
-                    am.setRingerMode(AudioManager.RINGER_MODE_VIBRATE);
-                } else {
-                    Utils.log("set ringer mode: silent");
-                    am.setRingerMode(AudioManager.RINGER_MODE_SILENT);
-                }
-                // refresh when event has ended
-                cursor.moveToFirst();
-                waitDuration = getWaitDuration(cursor.getString(3));
+            // if current lecture(s) found, silence the mobile
+            Utils.setInternalSetting(this, Const.SILENCE_ON, true);
+
+            // Set into silent mode
+            String mode = Utils.getSetting(this, "silent_mode_set_to", "0");
+            if ("0".equals(mode)) {
+                Utils.log("set ringer mode: vibration");
+                am.setRingerMode(AudioManager.RINGER_MODE_VIBRATE);
+            } else {
+                Utils.log("set ringer mode: silent");
+                am.setRingerMode(AudioManager.RINGER_MODE_SILENT);
             }
+            // refresh when event has ended
+            waitDuration = getWaitDuration(currentLectures.get(0).getDtstart());
```

Comment:
And this probably is handled better with an Optional of the current lecture instead of a List